client/ayon_core/tools/publisher/control.py

```diff
@@ -415,6 +464,52 @@ def _create_action_item(self, action, plugin_id):
             icon
         )
 
+    @staticmethod
+    def _filter_plugin_active_actions(actions, exception):
+        """ Filters active actions based on the plugin's specified conditions.
+
+        Determines which actions are active depending on the plugin's
+        specified conditions and the type and severity of the exception.
+
+        Args:
+            actions (List[Action]): A list of Action objects to filter.
+            exception (Type[Exception]): The exception raised during plugin
+            execution.
+
+        Returns:
+            List[Action]: A list of actions that are active based on the
+            specified conditions.
+
+        Raises:
+            None
+        """
+        plugin_errored = isinstance(exception, PublishValidationError)
+        error_is_blocking = getattr(exception, "is_blocking", False)
+
+        def is_action_active(action):
+            """ Evaluate if an action should be activated.
+
+            Checks whether an action meets its specified conditions for
+            activation, which can include being always active, or active in
+            response to specific states of the plugin.
+
+            Args:
+                action (Action): The action to evaluate.
+
+            Returns:
+                bool: Indicates if the action meets the activation criteria.
+            """
+            return action.active and (
+                    action.on == "all" or
+                    (action.on == "failedOrWarning" and plugin_errored) or
+                    (action.on == "failed" and
+                     (
+                             plugin_errored and error_is_blocking)
+                     )
+            )
+
+        return [action for action in actions if is_action_active(action)]
```

Comment:
for info: another syntax for this:

```suggestion
        return list(filter(is_action_active, actions))
```

Reply:
No impact on performance, it is just a matter of taste. Do as you please, it was for info only :)

Reply:
it's concise and clear enough. I like it :)